### DISCUSSION TITLE
Glossary Plugin

### DISCUSSION THREAD

**Discussion by AngryMane** [mkdocs/mkdocs#2542] [Q&A]
# Question 
Hi, I am a newbie to mkdoc. I'm using mkdocs to write software specification documents.
(mkdocs is so useful. Thank you for developing it!)
The text I am writing is large, and there are many project-specific words in it.
For this reason, I would like to implement the following measures to make the documents easier to understand

1. create a page for each project-specific word that explains the meaning of the word
1. link to the above page whenever a project-specific word appears in the documents.

I would like to automate the second steps because it would take a lot of time to do them manually. Is there any good way to do this?

# Background/What I'm doing now
I am trying to create a plugin that will do 1. and 2. automatically.
The image of the process of this plugin is as follows

a. Parse meta-page info(*1) of all pages, and make map information between project-specific words and their description pages
b. In markdown text, Replace project-specific words with a link to their description page

But I can't seem to get these processes to map well to the events that the current mkdocs plugin receives.

Than you!

(*1)   

```markdown
---
title: sample page
glossary: sample    # This declaration  means "This page is the description page of 'sample' word".
---

# Sample Page
snip...

```

**Comment by Andre601** [comment]
This isn't really a direct solution to your issue (I guess) but there exists the `abbreviations` extension which allows you to give definitions about specific words by defining them in the `*[text]: Description` format.

Material for MkDocs has a documentation page explaining this extension and how it can be used effectively with the snippets extension from PyMdownX.

https://squidfunk.github.io/mkdocs-material/reference/abbreviations/

This could allow you to write a collection of descriptions in a snippet to then include in the pages where necessary.

**Reply by AngryMane** [reply]
Thanks for letting me know. However, the words that need to be explained are still quite complex, so it is difficult to explain them with the abbreviations plugin. abbreviations plugin has the following problems
- The area where text can be written is very small
- The text can only be displayed while it is popup on the target word
